kapitan/refs/secrets/awskms.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"awskms secrets module"
+
+import base64
+import boto3
+import logging
+import warnings
+
+from kapitan.refs.base import Ref, RefBackend, RefError
+from kapitan import cached
+from kapitan.errors import KapitanError
+
+logger = logging.getLogger(__name__)
+
+# Ignore warning from googleapiclient if we are using default credentials
+warnings.filterwarnings("ignore", "Your application has authenticated using end user credentials")
+
+
+class AWSKMSError(KapitanError):
+    """Generic AWS KMS errors"""
+    pass
+
+
+def awskms_obj():
+    if not cached.awskms_obj:
+        # If --verbose is set, show requests from googleapiclient (which are actually logging.INFO)
```

Comment:
You can remove this if, verbosity is only required for googleapiclient

Reply:
done :+1: